modeshape-jcr/src/main/java/org/modeshape/jcr/value/binary/MongodbBinaryStore.java

```diff
@@ -192,8 +199,10 @@ public BinaryValue storeValue( InputStream stream,
 
             // store content
             DBCollection content = db.getCollection(key.toString());
-            ChunkOutputStream dbStream = markAsUnused ? new ChunkOutputStream(content, System.currentTimeMillis()) : new ChunkOutputStream(
-                                                                                                                                           content);
+            content.createIndex(DATA_CHUNK_SORT_INDEX);
```

Comment:
This method is called every time a value is stored. Is that required to add the document to the index? And is it safe to be called repeatedly? 

And IIUC, when this method does build the index it will be [done in the foreground](http://docs.mlab.com/indexing/). Is that correct? Would it be better to be done in the background?